apps/storefront-e2e/src/integration/cart.cy.ts

```diff
@@ -116,6 +143,160 @@ describe('Cart suite', () => {
           cy.checkGlobalNotificationAfterFailedApiCall(cartPage);
         });
       });
+
+      describe('with items in the cart and a valid coupon', () => {
+        beforeEach(() => {
+          user.addProductWithCoupon();
+          user.goToCart();
+        });
+
+        it('should update prices if the coupon is applied', () => {
+          cartPage.checkNotEmptyCart();
+          cartPage.getCartEntriesHeading().should('contain.text', '1 items');
+
+          cartPage.getCartTotals().checkTotals({
+            subTotal: '€366.00',
+            discountsTotal: '-€102.48',
+            taxTotal: '€42.07',
+            totalPrice: '€263.52',
+          });
+
+          cartPage.getCouponInput().type(coupon[0].code);
+          cartPage.getCouponBtn().click();
+
+          cartPage.getNotification().should('contain.text', '12345wu2ca');
+          cartPage
+            .getNotification()
+            .invoke('attr', 'type')
+            .should('eq', 'success');
+
+          cartPage.getCouponInput().should('have.value', '');
+
+          cartPage
+            .getCouponDate()
+            .shadow()
+            .should('contain.text', coupon[0].expiredDate);
+
+          cy.scrollTo('top');
+
+          cartPage.getCartTotals().checkTotals({
+            subTotal: '€366.00',
+            discountsTotal: '-€202.48',
+            taxTotal: '€26.11',
+            totalPrice: '€163.52',
+          });
+
+          if (user.userType === 'guest') {
```

Comment:
Does it make sense to make this if-else block if we do not perform login in these tests?